### PR TITLE
Custom DC: Add BT automation and initial reference data to install script

### PR DESCRIPTION
**Pull request by Fructokinase**
Clone tools repo and call BT automation Terraform module. Also copy from the reference folder in datcom-public bucket into a folder called "reference" in custom dc resource bucket. 

Note that the reference bucket is currently empty and will be filled with actual sample tmcf shortly.

### CODE REVIEW

File: scripts/install_custom_dc.sh [hunk 1]
```diff
@@ -83,6 +85,25 @@ terraform init && terraform apply \
   -var="project_id=$PROJECT_ID" \
   -var="dc_website_domain=$DOMAIN" -auto-approve
 
+# Run the BT automation Terraform script to set up BT loader.
+cd $ROOT
+git clone https://github.com/datacommonsorg/tools
+
+# TODO(alex): support custom robot SA and resource bucket name.
+WEBSITE_ROBOT="website-robot@$PROJECT_ID.google.com.iam.gserviceaccount.com"
+RESOURCE_BUCKET="$PROJECT_ID-resources"
```
Comment (shifucun) [inline scripts/install_custom_dc.sh:94]: do you need to create the bucket first?

**Comment by shifucun** [comment]
I tried the script.
1. There is a prompt to confirm, should skip that.
2. Got an error `Error: Error uploading object cloud_functions/bt_automation_go_source_chQTX3gEq8O6hb7ti7Qy3DJt6q3JRKJjmnx6AlPv4Qo=.zip: googleapi: Error 404: The specified bucket does not exist., notFound`

File: scripts/install_custom_dc.sh (hunk 1 above)
Comment (Fructokinase) [reply scripts/install_custom_dc.sh:94]: Please see the comment below.

**Comment by Fructokinase** [comment]
> I tried the script.
> 
> 1. There is a prompt to confirm, should skip that.
> 2. Got an error `Error: Error uploading object cloud_functions/bt_automation_go_source_chQTX3gEq8O6hb7ti7Qy3DJt6q3JRKJjmnx6AlPv4Qo=.zip: googleapi: Error 404: The specified bucket does not exist., notFound`

BT automation was originally going to be a part of setup. Seems like I removed the bucket when I moved Terraform module to tools repo. Added it back and updated the PR

**Review by shifucun — APPROVED** [review]

**Comment by shifucun** [comment]
Another error `googleapi: Error 403: Cloud Functions API has not been used in project 507955991373 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/cloudfunctions.googleapis.com/overview?project=507955991373 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.`

File: scripts/install_custom_dc.sh [hunk 2]
```diff
@@ -83,6 +85,26 @@ terraform init && terraform apply \
   -var="project_id=$PROJECT_ID" \
   -var="dc_website_domain=$DOMAIN" -auto-approve
 
+# Run the BT automation Terraform script to set up BT loader.
+cd $ROOT
+git clone https://github.com/datacommonsorg/tools
+
+# TODO(alex): support custom robot SA and resource bucket name.
+WEBSITE_ROBOT="website-robot@$PROJECT_ID.google.com.iam.gserviceaccount.com"
```
Comment (shifucun) [inline scripts/install_custom_dc.sh:93]: should remove ".google.com" in WEBSITE_ROBOT

**Comment by shifucun** [comment]
Another API issue `Error waiting for Creating CloudFunctions Function: Error code 7, message: Build failed: Cloud Build API has not been used in project 507955991373 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/cloudbuild.googleapis.com/overview?project=507955991373 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.`

File: scripts/install_custom_dc.sh [hunk 3]
```diff
@@ -83,6 +85,26 @@ terraform init && terraform apply \
...
+
+cd tools/bigtable_automation/terraform
+terraform init && terraform apply \
+  -var="project_id=$PROJECT_ID" \
+  -var="service_account_email=$WEBSITE_ROBOT" \
+  -var="dc_resource_bucket=$RESOURCE_BUCKET" \
+  -auto-approve
+
+# Copy over sample tmcfs/csvs from reference resource bucket.
+gsutil cp -r \
```
Comment (shifucun) [inline scripts/install_custom_dc.sh:104]: gsutil cp does not seem to work..

Comment (Fructokinase) [reply scripts/install_custom_dc.sh:104]: ack. If you mean entire folder copied over as opposed to files, folder levels are being copied over. As for the data, I will populate it shortly.

**Comment by Fructokinase** [comment]
Added permissions for datcom SA and also enabled cloud functions API. As for the publisher, I believe that will be be needed by the identity of the script that manually triggers the controller, so no need to set it here?

**Comment by shifucun** [comment]
> Added permissions for datcom SA and also enabled cloud functions API. As for the publisher, I believe that will be be needed by the identity of the script that manually triggers the controller, so no need to set it here?

Yes, sg.

**Pull request merged**